ledger/ledger-api-tests/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_8/ParticipantPruningIT.scala

```diff
@@ -838,4 +928,31 @@ class ParticipantPruningIT extends LedgerTestSuite {
         s"Active contract set comparison before and after pruning failed: $acsBeforePruning vs $acsAfterPruning",
       )
     }
+
+  // Note that the Daml template must be inspected to establish the key type and fields
+  // For the TextKey template the key is: (tkParty, tkKey) : (Party, Text)
+  // When populating the Record identifiers are not required.
+  private def makeTextKeyKey(party: ApiTypes.Party, keyText: String) = {
+    Value(
+      Value.Sum.Record(
+        Record(fields =
+          Vector(
+            RecordField(value = Some(Value(Value.Sum.Party(party.unwrap)))),
+            RecordField(value = Some(Value(Value.Sum.Text(keyText)))),
+          )
+        )
+      )
+    )
+  }
+
+  private def pruneToCurrentEnd(participant: ParticipantTestContext, party: Party)(implicit
+      ec: ExecutionContext
+  ): Future[LedgerOffset] = {
+    for {
+      end <- participant.currentEnd()
+      _ <- participant.create(party, Dummy(party)) // To increment the ledger end
```

Comment:
Hi @simonmaxen-da , this is unlikely to work against canton out-of-the-box because of canton ACS commitment reconciliation. There are two ways we can handle this both involving usage of `eventually`.

1. Wrap both the `participant.create` and `participant.prune` calls in eventually basically what we did in #16114 specially [here](https://github.com/digital-asset/daml/pull/16114/files#r1089031935)
2. Wrap only the `participant.prune` call in an eventually and configure canton's `LedgerApiParticipantPruningConformanceTest` to have the participants send each other ACS commitment "time proofs" say every second - a config not suitable for production but in test environments to make canton pruning behave more as you would expect. Let me know please if we choose to pursue this option.

I'd recommend running the test against canton locally to double-check this works.